app/models/base_aggregated_cash_transaction.rb

```diff
@@ -0,0 +1,169 @@
+      month_no = month_number(trx)
+      value_method = "#{trx.transaction_type.name}#{month_no}="
+      checkbox_method = "check_box_#{trx.transaction_type.name}="
+      model.__send__(value_method, trx.amount)
+      model.__send__(checkbox_method, 'true')
+    end
+
+    def month_number(trx)
+      current_month = trx.updated_at.to_date.at_beginning_of_month
+      MONTH_RANGE.each do |i|
+        return i if (current_month - i.months) == trx.transaction_date
+      end
+    end
+
+    def find_by(legal_aid_application_id:)
+      transactions = find_transactions(legal_aid_application_id)
+      model = new
+      transactions.each { |trx| populate_attribute(model, trx) }
+      model
+    end
+
+    def find_transactions(id)
+      transaction_type_ids = TransactionType.__send__(operation).map(&:id)
+
+      CashTransaction.where(
+        legal_aid_application_id: id,
+        transaction_type_id: transaction_type_ids
+      )
+    end
+  end
+
+  private_class_method :attributes_for_transaction_types,
+                       :populate_attribute,
+                       :month_number,
+                       :find_transactions
+
+  private
+
+  def update_attributes(params)
+    params.each do |key, value|
+      __send__("#{key}=", value)
+    end
+  end
+
+  def save_cash_transaction_records
+    self.class.cash_transaction_categories.each do |category|
+      CashTransaction.where(
+        legal_aid_application_id: legal_aid_application_id,
+        transaction_type_id: transaction_type_id(category)
+      ).destroy_all
+
+      save_category(category) if checkbox_for(category) == 'true'
+    end
+  end
+
+  def save_category(category)
+    MONTH_RANGE.each do |i|
+      amount = __send__("#{category}#{i}".to_sym)
+      date = calculated_date(i)
+
+      CashTransaction.create!(
+        legal_aid_application_id: legal_aid_application_id,
+        transaction_type_id: transaction_type_id(category),
+        transaction_date: date,
+        amount: amount
+      )
+    end
+  end
+
+  def checkbox_integrity
+    self.class.cash_transaction_categories.each do |category|
+      checkbox_attr = "check_box_#{category}".to_sym
+
+      if __send__(checkbox_attr) == 'true'
+        check_all_values_set_for category
+      else
+        erase_values_for category
+      end
+    end
+  end
+
+  def validate_at_least_one_selected
+    return if none_selected.present?
+
+    self.class.cash_transaction_categories.each do |category|
+      checkbox_attr = "check_box_#{category}".to_sym
+      return false if __send__(checkbox_attr) == 'true'
+    end
+
+    errors.add(:none_selected, model_error(:blank))
+  end
+
+  def validate_none_selected
+    return unless none_selected.present?
+
+    self.class.cash_transaction_categories.each do |category|
+      checkbox_attr = "check_box_#{category}".to_sym
+      next unless __send__(checkbox_attr) == 'true'
+
+      errors.add(:none_selected, model_error(:others_present))
```

Comment:
not sure what others_present does? I might be misunderstanding it.

my belief is that we have javascript around none_selected checkboxes to unselect other checkboxes and clear their values.

Reply:
Yes, it's belt and braces for non-js